src/configuration.js

```diff
@@ -41,30 +55,35 @@ export class Configuration {
 
     var rules = defaultRules;
 
-    if (this.options.only.length > 0) {
+    if (this.options.only && this.options.only.length > 0) {
       rules = filterRules(this.options.only);
-    } else if (this.options.except.length > 0) {
+    } else if (this.options.except && this.options.except.length > 0) {
       rules = defaultRules.filter(rule => {
         return (
           this.options.except.map(toUpperCamelCase).indexOf(rule.name) == -1
         );
       });
-    } else {
-      const directory = this.options.searchDirectory || process.cwd();
-      const cosmic = cosmiconfig('graphql-schema-linter', {
-        cache: false,
-        sync: true,
-      }).load(directory);
-
-      if (cosmic) {
-        rules = filterRules(cosmic.config.rules);
-      }
     }
 
     return rules;
   }
 }
 
+function loadOptionsFromConfig(directory) {
+  const cosmic = cosmiconfig('graphql-schema-linter', {
+    cache: false,
+    sync: true,
+  }).load(directory);
+  let options = {};
+  if (cosmic) {
+    // Map config.rules -> `only` param
```

Comment:
Awesome, yeah I think that will be cleaner.  I think we should do this in another PR for 2 reasons:

1. this is becoming a monster already 😆 
2. this would be a big breaking change and we should likely bump the major version number.